roachtest: Update asyncpg block list

spilchen opened the pull request:
This allows failures for some tests in the asyncpg tests that were due to the recent autocommit_before_ddl change in #141145. Rationale for each failing test are included as comments in the block list.

I will backport this in: https://github.com/cockroachdb/cockroach/pull/141987 https://github.com/cockroachdb/cockroach/pull/141851.

Epic: none
Release note: none
Closes #141878

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/142048)
<!-- Reviewable:end -->

spilchen committed:
roachtest: Update asyncpg block list

This allows failures for some tests in the asyncpg tests that were due
to the recent autocommit_before_ddl change in #141145. Rationale for
each failing test are included as comments in the block list.

Epic: none
Release note: none
Closes #141878

spilchen commented:
TFTR!

bors r+
